crypto/evp/encode.c

```diff
@@ -438,29 +438,81 @@ static int evp_decodeblock_int(EVP_ENCODE_CTX *ctx, unsigned char *t,
     if (n % 4 != 0)
         return -1;
 
-    for (i = 0; i < n; i += 4) {
-        a = conv_ascii2bin(*(f++), table);
-        b = conv_ascii2bin(*(f++), table);
-        c = conv_ascii2bin(*(f++), table);
-        d = conv_ascii2bin(*(f++), table);
-        if ((a & 0x80) || (b & 0x80) || (c & 0x80) || (d & 0x80))
-            return -1;
-        l = ((((unsigned long)a) << 18L) |
-             (((unsigned long)b) << 12L) |
-             (((unsigned long)c) << 6L) | (((unsigned long)d)));
-        *(t++) = (unsigned char)(l >> 16L) & 0xff;
-        *(t++) = (unsigned char)(l >> 8L) & 0xff;
-        *(t++) = (unsigned char)(l) & 0xff;
-        ret += 3;
+    if (ctx != NULL && (ctx->flags & EVP_ENCODE_CTX_DECODE_NO_PADDING) != 0) {
+        bp = n - 4;
+        for (i = 0; i < bp; i += 4) {
+            a = conv_ascii2bin(*(f++), table);
+            b = conv_ascii2bin(*(f++), table);
+            c = conv_ascii2bin(*(f++), table);
+            d = conv_ascii2bin(*(f++), table);
+            if ((a & 0x80) || (b & 0x80) || (c & 0x80) || (d & 0x80))
+                return -1;
+            l = ((((unsigned long)a) << 18L) |
+                 (((unsigned long)b) << 12L) |
+                 (((unsigned long)c) << 6L) | (((unsigned long)d)));
+            *(t++) = (unsigned char)(l >> 16L) & 0xff;
+            *(t++) = (unsigned char)(l >> 8L) & 0xff;
+            *(t++) = (unsigned char)(l) & 0xff;
+            ret += 3;
+        }
+
+        if (i < n) {
+            if ((*f == '=') || (*(f + 1) == '=') ||
+                ((*(f + 2) == '=') && (*(f + 3) != '=')))
+                return -1;
```

Comment:
it is somehow odd ot be picky about `=` characters here, but not in the previous loop.

Reply:
@bernd-edlinger Agree with you; we should do the same thing in the previous loop.